### PR TITLE
Prepare Inline Promotions

### PR DESCRIPTION
**Pull request by harryurban**
**WHAT**
- add and modify fields to be able to add inline elements in cigger

**WHY**
https://as-jira.axelspringer.de/browse/BER-6867

### CODE REVIEW

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala [hunk 1]
```diff
@@ -236,9 +236,12 @@ case class RawChannelConfiguration(metadata: Option[RawChannelMetadata] = None,
                                    theme: Option[RawChannelTheme] = None,
                                    commercial: RawChannelCommercial = RawChannelCommercial(),
                                    content: Option[RawChannelContentConfiguration] = None,
+                                   articlePromotions: Option[Seq[RawArticlePromotion]] = None,
```
Comment (aramin) [inline raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala]: parameter fehlt in den docs noch

File: core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala [hunk 2]
```diff
@@ -81,7 +81,7 @@ case class ApiContent(webUrl: String,
   lazy val unwrappedFields: Map[String, String] = fields.getOrElse(Map.empty[String, String])
   lazy val unwrappedAuthors: List[ApiAuthor] = authors.getOrElse(Nil)
   lazy val unwrappedElements: List[ApiElement] = elements.getOrElse(Nil)
-  lazy val unwrappedRoles: List[String] = roles.getOrElse(Nil)
+  def unwrappedRoles: List[String] = roles.getOrElse(Nil)
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala]: for mutable fields the convenience methods should be evaluated on each access, or else they contain the old value before mutation

File: core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala [hunk 3]
```diff
@@ -138,8 +138,8 @@ case class ApiAuthor(id: Option[String] = None,
 case class ApiElement(id: String,
                       `type`: String,
                       assets: Option[List[ApiAsset]],
-                      relations: Option[List[String]] = None) {
-  lazy val unwrappedRelations: List[String] = relations.getOrElse(Nil)
+                      var relations: Option[List[String]] = None) {
+  def unwrappedRelations: List[String] = relations.getOrElse(Nil)
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala]: have to mutate the original teaser element

again:
for mutable fields the convenience methods should be evaluated on each access, or else they contain the old value before mutation

File: core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala [hunk 4]
```diff
@@ -152,10 +152,10 @@ case class ApiElement(id: String,
   * @param index    The index for multiple assets like a gallery
   */
 case class ApiAsset(`type`: String,
-                    fields: Option[Map[String, String]] = None,
+                    var fields: Option[Map[String, String]] = None,
                     metadata: Option[ApiMetadata] = None,
                     index: Option[Int] = None) {
-  lazy val unwrappedFields: Map[String, String] = fields.getOrElse(Map.empty[String, String])
+  def unwrappedFields: Map[String, String] = fields.getOrElse(Map.empty[String, String])
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/models/ApiContent.scala]: have to mutate the fields to add the main content fields to the element fields to be able to teaser the main content as inline element

again:
for mutable fields the convenience methods should be evaluated on each access, or else they contain the old value before mutation

**Comment by harryurban** [comment]
@aramin / @zenderol 
are you cool with the changes?

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala (hunk 1 above)
Comment (harryurban) [reply raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala]: done

**Pull request merged**